File: backend/local-api-module/src/main/java/org/vyatsu/localApiModule/repository/ItemRepository.java
package org.vyatsu.localApiModule.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.vyatsu.localApiModule.dto.response.api.item.ItemDto;
import org.vyatsu.localApiModule.entity.item.Item;
import org.vyatsu.localApiModule.entity.user.User;

import java.util.List;

public interface ItemRepository extends JpaRepository<Item, Long> {
    List<Item> findByUserId(Long id);

    List<Item> findItemByUserAndIsDraft(User user, boolean isDraft);

    @Query("SELECT DISTINCT i.category FROM Item i WHERE i.user.id = ?1")
    List<String> findUniqueCategoryByUserId(Long id);

    Item findByVendorCode(String vendorCode);


    List<Item> findItemByIsDraft(Boolean isDraft);
}

File: app/src/main/java/com/bytebusters/android/mpsystem/repository/ItemRepository.kt
package com.bytebusters.android.mpsystem.repository

import com.bytebusters.android.mpsystem.entity.item.Item
import com.bytebusters.android.mpsystem.entity.user.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface ItemRepository : JpaRepository<Item?, Long?> {
    fun findByUserId(id: Long?): List<Item?>?
    fun findItemByUserAndIsDraft(user: User?, isDraft: Boolean): List<Item?>?

    @Query("SELECT DISTINCT i.category FROM Item i WHERE i.user.id = ?1")
    fun findUniqueCategoryByUserId(id: Long?): List<String?>?
    fun findByVendorCode(vendorCode: String?): Item?
    fun findItemByIsDraft(isDraft: Boolean?): List<Item?>?
}
